flocker/ca/_validation.py

```diff
@@ -27,16 +27,17 @@ class ControlServicePolicy(PRecord):
     :ivar Certificate ca_certificate: The certificate authority's
         certificate.
 
-    :ivar PrivateCertificate client_certificate: Client's
-        certificate/private key pair.
+    :ivar FlockerCredential client_credential: Client's certificate and
+        private key pair.
     """
     ca_certificate = field(mandatory=True)
-    client_certificate = field(mandatory=True)
+    client_credential = field(mandatory=True)
 
     def creatorForNetloc(self, hostname, port):
-        return optionsForClientTLS(u"control-service",
-                                   trustRoot=self.ca_certificate,
-                                   clientCertificate=self.client_certificate)
+        return optionsForClientTLS(
+            u"control-service",
+            trustRoot=self.ca_certificate,
+            clientCertificate=self.client_credential.private_certificate())
```

Comment:
I think that this says "Give me a ... (thing that's isomorphic to a TLS context) ... which requires the peer certificate to have a hostname that validates for `control-service` and has a signature chain from `self.ca_certificate` and present `self.client_credential.private_certificate()` as my key/certificate" - right?